common/src/main/scala/gu/msnotifications/NotificationHubClient.scala

```diff
@@ -22,33 +20,45 @@ final class NotificationHubClient(notificationHubConnection: NotificationHubConn
   import notificationHubConnection._
   import NotificationHubClient.HubResult
 
-  private def request(uri: String) = wsClient
-    .url(uri)
-    .withHeaders("Authorization" -> authorizationHeader(uri))
+  private def request(path: String) = {
+    val uri = s"$notificationsHubUrl$path?api-version=2015-01"
+    wsClient
+      .url(uri)
+      .withHeaders("Authorization" -> authorizationHeader(uri))
+  }
+
+  def register(registration: MobileRegistration): Future[HubResult[RegistrationResponse]] = {
+    register(RawWindowsRegistration.fromMobileRegistration(registration))
+  }
 
-  def register(rawWindowsRegistration: RawWindowsRegistration): Future[HubResult[RegistrationId]] = {
-    request(s"$notificationsHubUrl/registrations/?api-version=2015-01")
+  def register(rawWindowsRegistration: RawWindowsRegistration): Future[HubResult[RegistrationResponse]] = {
+    request("/registrations/")
       .post(rawWindowsRegistration.toXml)
-      .map(processRegistrationResponse)
+      .map(XmlParser.parse[RegistrationResponse])
   }
 
-  def update(registrationId: RegistrationId,
+  def update(registrationId: WNSRegistrationId,
              rawWindowsRegistration: RawWindowsRegistration
-              ): Future[HubResult[RegistrationId]] = {
-    request(s"$notificationsHubUrl/registration/${registrationId.registrationId}?api-version=2015-01")
+              ): Future[HubResult[RegistrationResponse]] = {
+    request(s"/registration/${registrationId.registrationId}")
       .post(rawWindowsRegistration.toXml)
-      .map(processRegistrationResponse)
+      .map(XmlParser.parse[RegistrationResponse])
   }
 
-  def sendPush(azureWindowsPush: AzureXmlPush): Future[HubResult[Unit]] = {
+  def sendNotification(azureWindowsPush: AzureXmlPush): Future[HubResult[Unit]] = {
     val serviceBusTags = azureWindowsPush.tagQuery.map(tagQuery => "ServiceBusNotification-Tags" -> tagQuery).toList
 
-    request(s"$notificationsHubUrl/messages/?api-version=2015-01")
+    request(s"/messages/")
       .withHeaders("X-WNS-Type" -> azureWindowsPush.wnsType)
       .withHeaders("ServiceBusNotification-Format" -> "windows")
       .withHeaders(serviceBusTags: _*)
       .post(azureWindowsPush.xml)
-      .map(processResponse).map(_.map(_ => ()))
+      .map { response =>
+        if (response.status == 200)
+          \/.right(())
+        else
+          \/.left(HubFailure.fromResponseCode(response.status))
```

Comment:
I would play safe and re-use the status message from the response instead of deducing it from the code. I know it's supposed to be fairly standard but you never know.